src/transaction-runner.coffee

```diff
@@ -66,8 +66,22 @@ class TransactionRunner
 
         catch error
           if error instanceof chai.AssertionError
-            data.message = "Failed assertion in hooks: " + error.message
-            data.test?.status = 'fail'
+            message = "Failed assertion in hooks: " + error.message
+
+            data['results'] ?= {}
+            data['results']['general'] ?= []
+            data['results']['general'].push { severity: 'error', message: message }
+
+            data.message = message
+
+            data['test'] ?= {}
+            data['test']['status'] = 'fail'
+
+            data['test']['results'] ?= {}
+
+            for key, value of data.results
+              data['test']['results'][key] = value
```

Comment:
is this duplicating a lot of things onto just another object? Who has introduced that and why?

![oh my](http://media.giphy.com/media/3MQVN9JzWSFt6/giphy.gif)